collect_app/src/main/java/org/odk/collect/android/activities/FormFillingActivity.java

```diff
@@ -1792,23 +1791,12 @@ protected void onResume() {
                 if (fec != null) {
                     loadingComplete(formLoaderTask, formLoaderTask.getFormDef(), null);
                 } else {
-                    DialogFragmentUtils.dismissDialog(FormLoadingDialogFragment.class, getSupportFragmentManager());
-                    FormLoaderTask t = formLoaderTask;
-                    formLoaderTask = null;
-                    t.cancel();
-                    t.destroy();
-                    // there is no formController -- fire MainMenu activity?
-                    Timber.w("Starting MainMenuActivity because formController is null/formLoaderTask not null");
-                    startActivity(new Intent(this, MainMenuActivity.class));
+                    throw new IllegalStateException("Null formController!");
```

Comment:
This (and the below case) don't have tests and as far as I'm aware, we don't know how they happen. Crashing means we don't have to deal with form session/lock clean up, and I actually think it's a better experience now that we show the crash message on the next relaunch than mysteriously returning to the Main Menu without warning.